src/tools/error_index_generator/build.rs

```diff
@@ -54,4 +54,9 @@ fn register_all() -> Vec<(&'static str, Option<&'static str>)> {
     all.push_str("}\n");
 
     fs::write(&dest, all).unwrap();
+
+    // We link against `rustc_driver` so we may need the LLVM library to get
+    // linked.
+    println!("cargo:rerun-if-env-changed=CFG_LLVM_LIBDIR");
+    println!("cargo:rustc-link-search=native={}", env!("CFG_LLVM_LIBDIR"));
```

Comment:
This is unfortunately due to how dylib linkage works where it's passed as `-lfoo` to all linked complications after the first (due to monomorphizations perhaps introducing new symbols). As a result everyone who links to `rustc_driver` will need to find the dylib.

Reply:
This will complicate things for everyone using rustc_driver and for projects outside of rust-lang/rust `env!("CFG_LLVM_LIBDIR")` is not defined.

Reply:
Can we work around this by having a shim crate around rustc driver which has this bit in it? It sounds like that probably won't work based on your explanation, but I'm not quite sure. I do think it'd be good to figure out the story for out-of-tree builds (e.g., clippy CI), though it can be kinda not great I think.

Reply:
A crate can't really solve this one way or another unfortunately, the problem is how rustc propagates linked libraries and linked flags. 

I'll look to see if we can solve this with `static-nobundle` one way or another.